regression/pages/common/utils.py

```diff
@@ -0,0 +1,526 @@
+    AUTH_USERNAME,
+    URL_WITHOUT_AUTH,
+    DEFAULT_TIMEOUT,
+    ECOMMERCE_URL_WITHOUT_AUTH,
+    ECOMMERCE_API_URL,
+    EMAIL_SENDER_ACCOUNT,
+    INITIAL_WAIT_TIME,
+    ORG,
+    TIME_OUT_LIMIT,
+    WAIT_TIME
+)
+
+
+class MailException(Exception):
+    """
+    Exceptions for Mail failures
+    """
+    pass
+
+
+class ApiException(Exception):
+    """
+    Exceptions for API failures
+    """
+    pass
+
+
+class GuerrillaMailApi(object):
+    """
+    Guerrilla Mail API for creating email accounts and reading emails
+    """
+
+    def __init__(self):
+        self.session = GuerrillaMailSession()
+
+    def get_email_account(self, user_name):
+        """
+        This function will create an account on GuerrillaMail using given user
+        name and first of the available domains. The email address created by
+        joining user name and domain is returned
+        Args:
+            user_name:
+        Returns:
+            email address:
+        """
+        self.session.set_email_address(user_name)
+        return self.session.get_session_state()['email_address']
+
+    def get_email_text(self, pattern):
+        """
+        This function will check the availability MIT enrollment email at
+        GuerrillaMail server at the intervals of 5 seconds and if the email
+        is found it's text is returned. If the email is not found after 40
+        seconds, this function will raise and error
+        Args:
+            pattern:
+        Returns:
+            email text:
+        """
+        email_text = ""
+        t_end = time.time() + TIME_OUT_LIMIT
+        # Run the loop for a pre defined time
+        time.sleep(INITIAL_WAIT_TIME)
+        while time.time() < t_end:
+            try:
+                # Check that mail box is not empty
+                email_list = self.session.get_email_list()
+                if not email_list:
+                    # raise an exception that waits 3 seconds before
+                    # restarting loop
+                    raise MailException
+                # Get the email id of last email in the inbox
+                email_id = email_list[0].guid
+                # if last email is not sent by MIT raise the exception
+                email = self.session.get_email(email_id)
+                if email.sender != EMAIL_SENDER_ACCOUNT[ORG]:
+                    raise MailException
+                # Fetch the email text and stop the loop
+                email_text = email.body
+                if pattern not in email_text:
+                    raise MailException
+                break
+            except MailException:
+                time.sleep(WAIT_TIME)
+        if email_text:
+            return email_text
+        else:
+            raise MailException('No Email from ' + EMAIL_SENDER_ACCOUNT[ORG])
+
+
+class TempMailApi(object):
+    """
+    Temp Mail API for creating email accounts and reading emails
+    """
+
+    def __init__(self):
+        self.session = TempMail()
```

Comment:
same here...and in fact, seeing `self.session` twice in the same file, meaning different things...that highlights my concern. Let's be more specific with it.